kernelci/build.py

```diff
@@ -341,6 +356,20 @@ def push_tarball(config, kdir, storage, api, token):
     return tarball_url
 
 
+def pull_tarball(kdir, url, dest_filename, retries, retry_sleep):
+    recreate_dir(kdir)
+    while retries > 0:
```

Comment:
Consider something like `for i in range(0, retry_sleep)`, so that you don't need to modify retry_sleep (you can just `sleep(i)`), which I think is a bad idea. It also means the sleep will get longer each retry, rather than shorter. It may be better to have even more backoff, such as `sleep(2**i)`.

Reply:
Sorry, I now notice retries and retry_sleep which I conflated in my first message. My concrete suggestion is to drop retry_sleep, and do `for i in range(1, retries): sleep(2**i)`.

Reply:
It looks like a good idea. I modified the implementation to gradually increase the sleep time in case of consequent failures.